### PR TITLE
lxd: Stop setting $HOME in containers

### PR DESCRIPTION
**Pull request by kalikiana**
Because the project folder is mounted into the container and `$HOME` points at it, .cache ends up in the project folder. We need to stop setting `$HOME`.

This addresses one aspect of bug [1703642](https://bugs.launchpad.net/snapcraft/+bug/1703642).

### CODE REVIEW

**Comment by elopio** [comment]
Saviq left a question in the bug, please reply that one. And I have another.

Why are we setting home to be the mounted project directory? 

I find this can cause more confusions along the way. If I want to debug my container, I would expect the cache to be in /root/.cache, I wouldn't think about finding it in /run/.

So why don't we leave $HOME to /root, just not touching it. Then we cd into /project-dir, or even /root/project-dir to execute the snapcraft command? I might be missing something obvious here, but that's what I would look for if I bash into the container to debug.

**Comment by Saviq** [comment]
A good first step :)

File: snapcraft/internal/lxd.py [hunk 1]
```diff
@@ -74,16 +75,20 @@ def _get_remote_info(self):
...
-        check_call(['lxc', 'exec', self._container_name, '--env',
-                   'HOME=/{}'.format(self._project_folder), '--'] + cmd)
+    def _container_run(self, cmd, cwd=None):
+        if cwd:
+            check_call(['lxc', 'exec', self._container_name, '--',
+                        'bash', '-c', 'cd {}; {}'.format(
+                            cwd,
+                            ' '.join(pipes.quote(arg) for arg in cmd))])
+        else:
+            check_call(['lxc', 'exec', self._container_name, '--'] + cmd)
```
Comment (elopio) [inline snapcraft/internal/lxd.py]: I would like this more with only one check_call. Like:
```
if cwd:
    cmd = ['bash', '-c', 'cd {}; {}'.format(cwd, ...)]
check_call(['lxc', ...] + cmd)
```

File: snapcraft/internal/lxd.py [hunk 2]
```diff
@@ -137,7 +142,8 @@ def _setup_project(self):
                            os.path.basename(tar_filename))
         self._container_run(['mkdir', self._project_folder])
         self._push_file(tar_filename, dst)
-        self._container_run(['tar', 'xvf', os.path.basename(tar_filename)])
+        self._container_run(['tar', 'xvf', os.path.basename(tar_filename)],
+                            cwd=self._project_folder)
```
Comment (elopio) [inline snapcraft/internal/lxd.py]: I learned that in the upcoming lxc version, it's possible to just push the directory.
Not very useful though, as we have to support the version in trusty. Just fyi.

**Pull request merged**